### PR TITLE
[FIX] Apps converters delete fields on message attachments

### PR DESCRIPTION
**Pull request by d-gubert**
Whenever a message is sent over to the apps engine to trigger a hook, the message converter that bridges the data from Rocket.Chat to the Engine does not take all the fields from the original object to the converted one.

When data comes back from the Engine, it was not being merged back to the message object properly, which was causing the attachments (and any other non-primitive value) to lose some of the fields.

With these changes, fields that are ignored by the Engine will no longer be deleted from the message's attachments.

Closes https://github.com/RocketChat/Rocket.Chat/issues/12691

### CODE REVIEW

**Comment by rodrigok** [comment]
@d-gubert how this handle fields removal? If I unset, like, attachments from inside the engine?

**Comment by rodrigok** [comment]
> @d-gubert how this handle fields removal? If I unset, like, attachments from inside the engine?

This may answer my question:
https://github.com/RocketChat/Rocket.Chat/pull/14028/files#diff-c961b69f32e88f90d70fea25dc2ae0b2R4

File: app/utils/lib/merge.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (rodrigok) [inline app/utils/lib/merge.js]: Would be really nice to have some unit tests here

**Comment by d-gubert** [comment]
Discussing with @sampaiodiego and @tassoevan we found a more elegant solution. I'll be closing this PR for now until I have implemented it.

**Comment by rodrigok** [comment]
The code looks good, just missing some unit tests 😄

**Comment by d-gubert** [comment]
Added some tests.

Some of the mocks are kind of generic and maybe could be moved to a better place? What do you think @rodrigok? 🤔

**Pull request merged**